java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/PromoService.java

```diff
@@ -99,7 +101,7 @@ private String getLastUsers(PromoCampaignRepository.Promo promoCampaign) {
 		return emails.substring(0, emails.length() - 3);
 	}
 
-	public PromoResponse createPromoSubscription(String email, String key, Date expireTime) {
+	public PromoResponse createPromoSubscription(String email, String key, Date expireTime) throws FileNotFoundException, UnsupportedEncodingException {
```

Comment:
Exceptions shuold be RuntimeException by guidelines as we don't need here checked exceptions